shcDecoder/src/main/java/ca/bc/gov/shcdecoder/cache/impl/FileManagerImpl.kt

```diff
@@ -59,29 +63,51 @@ class FileManagerImpl(
     }
 
     override suspend fun getIssuers(url: String): List<Issuer> {
-        val jsonString = getJsonStringFromFile(url)
-        val issuerResponse = Gson().fromJson(jsonString, TrustedIssuersResponse::class.java)
-        return issuerResponse.trustedIssuers
+        val issuerResponse = getDataFromFile(url, TrustedIssuersResponse::class.java)
+        return issuerResponse?.trustedIssuers.orEmpty()
     }
 
     override suspend fun getKeys(url: String): List<JwksKey> {
-        val jsonString = getJsonStringFromFile(url)
-        val keyResponse = Gson().fromJson(jsonString, Jwks::class.java)
-        return keyResponse.keys
+        val keyResponse = getDataFromFile(url, Jwks::class.java)
+        return keyResponse?.keys.orEmpty()
     }
 
     override suspend fun getRule(url: String): List<Rule> {
-        val jsonString = getJsonStringFromFile(url)
-        val validationRulesResponse =
-            Gson().fromJson(jsonString, ValidationRuleResponse::class.java)
-        return validationRulesResponse.ruleSet
+        val validationRulesResponse = getDataFromFile(url, ValidationRuleResponse::class.java)
+        return validationRulesResponse?.ruleSet.orEmpty()
     }
 
-    private fun getJsonStringFromFile(url: String): String {
-        val fileName = getFileNameFromUrl(url)
-        val file = File(downloadDir, fileName)
-        val bufferedReader = file.bufferedReader()
-        return bufferedReader.use { it.readText() }
+    override suspend fun getRevocations(url: String): List<Pair<String, Date?>> {
```

Comment:
getRecovations & getRevocationsCtr can be clubbed in to one method with a custom model class that hold both this return type in single data class.